packages/js-dpp/lib/dataContract/stateTransition/DataContractUpdateTransition/validation/basic/validateDataContractUpdateTransitionBasicFactory.js

```diff
@@ -0,0 +1,168 @@
+
+const Identifier = require('../../../../../identifier/Identifier');
+
+/**
+ * @param {JsonSchemaValidator} jsonSchemaValidator
+ * @param {validateDataContract} validateDataContract
+ * @param {validateProtocolVersion} validateProtocolVersion
+ * @param {StateRepository} stateRepository
+ * @param {DiffValidator} diffValidator
+ * @param {validateIndicesAreNotChanged} validateIndicesAreNotChanged
+ * @param {JsonPatch} jsonPatch
+ *
+ * @return {validateDataContractUpdateTransitionBasic}
+ */
+function validateDataContractUpdateTransitionBasicFactory(
+  jsonSchemaValidator,
+  validateDataContract,
+  validateProtocolVersion,
+  stateRepository,
+  diffValidator,
+  validateIndicesAreNotChanged,
+  jsonPatch,
+) {
+  /**
+   * @typedef validateDataContractUpdateTransitionBasic
+   * @param {RawDataContractUpdateTransition} rawStateTransition
+   * @return {ValidationResult}
+   */
+  async function validateDataContractUpdateTransitionBasic(rawStateTransition) {
+    const result = jsonSchemaValidator.validate(
+      dataContractUpdateTransitionSchema,
+      convertBuffersToArrays(rawStateTransition),
+    );
+
+    if (!result.isValid()) {
+      return result;
+    }
+
+    result.merge(
+      validateProtocolVersion(rawStateTransition.protocolVersion),
+    );
+
+    if (!result.isValid()) {
+      return result;
+    }
+
+    // Validate Data Contract
+    const rawDataContract = rawStateTransition.dataContract;
+
+    result.merge(
+      await validateDataContract(rawDataContract),
+    );
+
+    if (!result.isValid()) {
+      return result;
+    }
+
+    const dataContractId = Identifier.from(rawDataContract.$id);
+
+    // Data contract should exist
+    const existingDataContract = await stateRepository.fetchDataContract(dataContractId);
+
+    if (!existingDataContract) {
+      result.addError(
+        new DataContractNotPresentError(dataContractId.toBuffer()),
+      );
+
+      return result;
+    }
+
+    // Version difference should be exactly 1
+    const oldVersion = existingDataContract.getVersion();
+    const newVersion = rawDataContract.version;
+    const versionDiff = newVersion - oldVersion;
+
+    if (versionDiff !== 1) {
+      result.addError(
+        new InvalidDataContractVersionError(
+          oldVersion + 1,
+          oldVersion + versionDiff,
+        ),
+      );
+    }
+
+    // Schema should be backward compatible
+    const oldSchema = existingDataContract.getDocuments();
+    const newSchema = rawDataContract.documents;
+
+    try {
+      diffValidator.validateSchemaCompatibility(oldSchema, newSchema);
+    } catch (e) {
+      const error = new IncompatibleDataContractSchemaError(
+        existingDataContract.getId().toBuffer(),
+      );
+      error.setOldSchema(oldSchema);
+      error.setNewSchema(newSchema);
+      error.setValidationError(e);
+
+      result.addError(error);
+
+      return result;
+    }
+
+    // check that only $defs, version and documents are changed
```

Comment:
I have a feeling that this check is better to do before we check backward compatibility